Formula/n/netatalk.rb

```diff
@@ -0,0 +1,54 @@
+class Netatalk < Formula
+  desc "File server for Macs, compliant with Apple Filing Protocol (AFP)"
+  homepage "https://netatalk.io"
+  url "https://github.com/Netatalk/netatalk/releases/download/netatalk-4-0-8/netatalk-4.0.8.tar.xz"
+  sha256 "d09d591521b27b91b1c2a6255a2e059af8c9fda43570f983f0a145706e5f7628"
+  license all_of: [
+    "GPL-2.0-only",
+    "GPL-2.0-or-later",
+    "LGPL-2.0-only",
+    "LGPL-2.1-or-later",
+    "BSD-2-Clause",
+    "BSD-3-Clause",
+    "MIT",
+  ]
+  head "https://github.com/Netatalk/netatalk.git", branch: "main"
+
+  depends_on "docbook-xsl" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkgconf" => :build
+
+  depends_on "berkeley-db" # macOS bdb library lacks DBC type etc.
```

Comment:
Probably need to use `berkeley-db@5` for compatibility with GPLv2 part of license. `berkeley-db` is newer AGPLv3 version.

Reply:
We've had to change other formulae for same reason, so agreed.

Reply:
Thanks for the advice! I have made this change now.

Reply:
When using berkeley-db@5, the brew linkage validation fails on Linux specifically. I think `libdb-5.3.so` is very much wanted. Can this be ignored?

```
brew linkage --test netatalk
  Unwanted system libraries:
    /lib/x86_64-linux-gnu/libdb-5.3.so
```

Reply:
Sounds like an RPATH issue. `brew` should automatically add the path to `berkeley-db@5` in RPATH.

Is the build script adding its own RPATH? May also need to inspect ELF binary to see what is RPATH.

Reply:
Thanks for the tip. The netatalk build system does indeed add its own RPATH by default. I have now injected `-Dwith-rpath=false` to the Meson arguments. Unfortunately, this change by itself did not have any impact on the issue at hand, so more troubleshooting is required...

Reply:
Looking at previous run artifacts, part of `brew` RPATH is added, but the `berkeley-db@5` is oddly missing:
```console
❯ objdump -p sbin/cnid_dbd | grep RPATH | awk '{print $2}' | tr ':' '\n'
@@HOMEBREW_PREFIX@@/Cellar/netatalk/4.0.8/lib
@@HOMEBREW_PREFIX@@/opt/gcc/lib/gcc/current
@@HOMEBREW_PREFIX@@/opt/openssl@3/lib
@@HOMEBREW_PREFIX@@/opt/libevent/lib
@@HOMEBREW_PREFIX@@/opt/libgpg-error/lib
@@HOMEBREW_PREFIX@@/opt/libgcrypt/lib
@@HOMEBREW_PREFIX@@/opt/abseil/lib
@@HOMEBREW_PREFIX@@/opt/icu4c@76/lib
@@HOMEBREW_PREFIX@@/opt/lz4/lib
@@HOMEBREW_PREFIX@@/opt/zlib/lib
@@HOMEBREW_PREFIX@@/opt/protobuf/lib
@@HOMEBREW_PREFIX@@/opt/xz/lib
@@HOMEBREW_PREFIX@@/opt/zstd/lib
@@HOMEBREW_PREFIX@@/opt/brotli/lib
@@HOMEBREW_PREFIX@@/opt/libnghttp2/lib
@@HOMEBREW_PREFIX@@/opt/libssh2/lib
@@HOMEBREW_PREFIX@@/opt/rtmpdump/lib
@@HOMEBREW_PREFIX@@/opt/ncurses/lib
@@HOMEBREW_PREFIX@@/opt/libedit/lib
@@HOMEBREW_PREFIX@@/opt/krb5/lib
@@HOMEBREW_PREFIX@@/opt/libxcrypt/lib
@@HOMEBREW_PREFIX@@/opt/readline/lib
@@HOMEBREW_PREFIX@@/opt/sqlite/lib
@@HOMEBREW_PREFIX@@/opt/util-linux/lib
@@HOMEBREW_PREFIX@@/opt/openldap/lib
@@HOMEBREW_PREFIX@@/opt/libunistring/lib
@@HOMEBREW_PREFIX@@/opt/libidn2/lib
@@HOMEBREW_PREFIX@@/opt/curl/lib
@@HOMEBREW_PREFIX@@/opt/cyrus-sasl/lib
@@HOMEBREW_PREFIX@@/opt/libtirpc/lib
@@HOMEBREW_PREFIX@@/opt/mysql/lib
@@HOMEBREW_PREFIX@@/opt/gdbm/lib
@@HOMEBREW_PREFIX@@/opt/expat/lib
@@HOMEBREW_PREFIX@@/opt/perl/lib
@@HOMEBREW_PREFIX@@/lib
```

Not sure if one of earlier RPATH changes are interfering. The arguments that superenv pass can be seen from logs file (02.meson.cc)
```
-Wl,-rpath,$ORIGIN/../../libatalk:/home/linuxbrew/.linuxbrew/Cellar/mysql/9.0.1_9/lib:/home/linuxbrew/.linuxbrew/opt/berkeley-db@5/lib:/home/linuxbrew/.linuxbrew/Cellar/libtirpc/1.3.6/lib:/home/linuxbrew/.linuxbrew/Cellar/netatalk/4.0.8/lib
-Wl,-rpath-link,/tmp/netatalk-20241230-6923-xkpibf/netatalk-4.0.8/build/libatalk
-Wl,-rpath-link,/home/linuxbrew/.linuxbrew/Cellar/mysql/9.0.1_9/lib
-Wl,-rpath-link,/home/linuxbrew/.linuxbrew/opt/berkeley-db@5/lib
-Wl,-rpath-link,/home/linuxbrew/.linuxbrew/Cellar/libtirpc/1.3.6/lib
...
-Wl,--end-group
...
-Wl,-rpath=/home/linuxbrew/.linuxbrew/Cellar/netatalk/4.0.8/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/gcc/lib/gcc/current
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/berkeley-db@5/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/openssl@3/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/libevent/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/libgpg-error/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/libgcrypt/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/abseil/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/icu4c@76/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/lz4/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/zlib/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/protobuf/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/xz/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/zstd/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/brotli/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/libnghttp2/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/libssh2/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/rtmpdump/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/ncurses/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/libedit/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/krb5/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/libxcrypt/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/readline/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/sqlite/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/util-linux/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/openldap/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/libunistring/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/libidn2/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/curl/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/cyrus-sasl/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/libtirpc/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/mysql/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/gdbm/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/expat/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/opt/perl/lib
-Wl,-rpath=/home/linuxbrew/.linuxbrew/lib
-Wl,--dynamic-linker=/home/linuxbrew/.linuxbrew/lib/ld.so
```

Reply:
This was a valuable lead, thank you. I think I have narrowed down the problem to a bug in the netatalk build scripts. The linking works properly in my local Linux environment now, but I am running into another problem in the Homebrew CI environment that I'm dealing with.